sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DoNotHardcodeCredentialsTest.cs

```diff
@@ -43,7 +43,7 @@ public void DoNotHardcodeCredentials_CS_DefaultValues()
         public void DoNotHardcodeCredentials_CS_CustomValues()
         {
             Verifier.VerifyAnalyzer(@"TestCases\DoNotHardcodeCredentials_CustomValues.cs",
-                new CSharp.DoNotHardcodeCredentials(AnalyzerConfiguration.AlwaysEnabled) { CredentialWords = "kode,facal-faire" });
+                new CSharp.DoNotHardcodeCredentials(AnalyzerConfiguration.AlwaysEnabled) { CredentialWords = @"kode,facal-faire,\*+?|{}[]()^$.# " });
```

Comment:
The regex is not valid, but {}, [] and () are quite close to valid syntax. I suggest to swap them to be absolutely invalid RegExes on first sight.
```suggestion
                new CSharp.DoNotHardcodeCredentials(AnalyzerConfiguration.AlwaysEnabled) { CredentialWords = @"kode,facal-faire,\*+?|}{][)(^$.# " });
```